File: AtividadeServidor/FrontEnd/app/src/main/java/com/ufc/navegacaoentretelas/model/Carro.java
package com.ufc.navegacaoentretelas.model;

public class Carro {

    private static int contadorId = 0;

    private int id;
    private String nome;
    private String marca;
    private String placa;
    private String ano;

    public Carro(String nome, String marca, String placa, String ano) {

        this.id = contadorId++;

        this.nome = nome;
        this.marca = marca;
        this.placa = placa;
        this.ano = ano;
    }

    public int getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public String getPlaca() {
        return placa;
    }

    public void setPlaca(String placa) {
        this.placa = placa;
    }

    public String getAno() {
        return ano;
    }

    public void setAno(String ano) {
        this.ano = ano;
    }

    @Override
    public String toString() {
        return nome;
    }
}

File: AtividadeServidor/BackEnd/app/models/carro.js
import mongoose from 'mongoose'

function carros(){
    let schema = mongoose.Schema({
        nome:{
            type: String,
            require: true
        },
        marca:{
            type: String,
            require: true,
            unique: true
        },
        placa:{
            type: String,
            require: true
        },
        ano:{
            type: String,
            require: true
        }
    })
    return mongoose.model("Carro",schema)
}

export default carros()